test/E2E/E2ETest.cs

```diff
@@ -50,6 +51,78 @@ public void TestDotnetCompile()
             TestOutputExecutable(OutputDirectory);
         }
 
+        [Fact] 
+        public void TestDotnetIncrementalBuild()
+        {
+            TestSetup();
+
+            //first build
+            TestRunCommand("dotnet", $" --verbose build --build-profile -o {OutputDirectory}");
```

Comment:
Please update these tests to use the Test infrastructure similar to the new factoring in E2E